doc/pipeline_spec.md

```diff
@@ -43,17 +43,92 @@
 
 `inputs` specifies a set of Repos that will be visible to the jobs during runtime. Commits to these repos will automatically trigger the pipeline to create new jobs to process them.
 
-`inputs.method` specifies how a repo will be partitioned among parallel containers, and whether the entire repo or just the new commit is used as the input.
+`inputs.method` specifies two different properties:
+- Partition unit: How input data  will be partitioned across parallel containers.
+- Incrementality: Whether the entire all of the data or just the new data (diff) is processed. 
+ 
+The next section explains input methods in detail.
 
-You may specify a method using either an alias or a JSON object.  We support four aliases that represent the four commonly used strategies:
+### Pipeline Input Methods
 
-* map: each job sees a part of the new commit; files may be partitioned.
-* reduce: each job sees a part of the entire repo; files are not partitioned
-* incremental_reduce: each job sees a part of the new commit; files are not partitioned.
-* global: each job sees the entire repo
+For each pipeline input, you may specify a "method".  A method dictates exactly what happens in the pipeline when a commit comes into the input repo.
 
-If a method is not specified, the "map" method is used by default.
+A method consists of two properties: partition unit and incrementality.
 
+#### Partition Unit
+Partition unit specifies the granularity at which input data is parallelized across containers.  
+
+* `Blocks`: different blocks of the same file may be parelleized across containers.
+* `Top-level objects`: the files and/or directories residing under the root directory (/) must be grouped together.  For instance, if you have four files in a directory structure like: 
```

Comment:
Yeah I feel the same way, but @jdoliner and I talked about this a bit and we can't think of any other options at the moment.